### PR TITLE
Cleanup of the clone meeting code.

### PR DESCRIPTION
**Pull request by reiterl**
Use the import merged user feature to cleanup here.

### CODE REVIEW

**Review by r-peschke — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/meeting/clone.py [hunk 1]
```diff
@@ -75,11 +75,15 @@ def preprocess_data(self, instance: Dict[str, Any]) -> Dict[str, Any]:
...
         meeting_json = export_meeting(self.datastore, instance["meeting_id"])
         instance["meeting"] = meeting_json
-        additional_user_ids = instance.pop("user_ids", None)
-        additional_admin_ids = instance.pop("admin_ids", None)
+        self.additional_user_ids = instance.pop("user_ids", None) or []
+        self.additional_admin_ids = instance.pop("admin_ids", None) or []
 
         # needs an empty map for superclass code
         self.merge_user_map = {}
+        for id_ in self.get_meeting_from_json(meeting_json).get("user_ids", []):
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting/clone.py:83]: *Wrong*. You need to handle the users with merge user mechanism.

File: tests/system/action/meeting/test_clone.py [hunk 2]
```diff
@@ -144,6 +144,8 @@ def test_clone_with_users(self) -> None:
         self.set_models(self.test_models)
         response = self.request("meeting.clone", {"meeting_id": 1})
         self.assert_status_code(response, 200)
+        self.assert_model_exists("meeting/1", {"user_ids": [1]})
```
[the remark this answers is not in the document]
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:147]: but what? What should I do here?

[the remark this answers is not in the document]
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:147]: `test_clone_user_ids_and_admin_ids` is the test for user_ids and admin_ids and groups, meeting.

[the remark this answers is not in the document]
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:147]: committee are just set and not added.

File: openslides_backend/action/actions/meeting/clone.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (r-peschke) [reply openslides_backend/action/actions/meeting/clone.py:83]: What do you mean? You don't need to search for usernames/email... and/or create/duplicate users

File: tests/system/action/meeting/test_clone.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (r-peschke) [reply tests/system/action/meeting/test_clone.py:147]: > but what? What should I do here?

Write a test for the user being member of a new committee, the relation between cloned meeting  and committee, organization. These are the problematic calculated fields.

**Comment by reiterl** [comment]
Rebase to a current main.

File: openslides_backend/action/actions/meeting/clone.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting/clone.py:83]: I mean, that I need to fill the merge_user_map and use the underlying mechanisms.

[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting/clone.py:83]: I am not sure, what you mean with your "you don't need to merge the users"...

File: tests/system/action/meeting/test_clone.py [hunk 3]
```diff
@@ -259,11 +261,45 @@ def test_clone_user_ids_and_admin_ids(self) -> None:
...
             },
         )
 
+    def test_clone_new_committee_and_add_user(self) -> None:
+        self.set_models(self.test_models)
+        self.set_models(
+            {
+                "user/13": {
+                    "username": "user_from_new_committee",
+                    "committee_ids": [2],
```
Comment (r-peschke) [inline tests/system/action/meeting/test_clone.py:275]: Remove this line

File: tests/system/action/meeting/test_clone.py [hunk 4]
```diff
@@ -259,11 +261,45 @@ def test_clone_user_ids_and_admin_ids(self) -> None:
...
+                },
+                "committee/2": {"organization_id": 1, "user_ids": [13]},
```
Comment (r-peschke) [inline tests/system/action/meeting/test_clone.py:277]: remove field user_ids from this line

File: tests/system/action/meeting/test_clone.py [hunk 5]
```diff
@@ -259,11 +261,45 @@ def test_clone_user_ids_and_admin_ids(self) -> None:
...
+        response = self.request(
+            "meeting.clone",
+            {
+                "meeting_id": 1,
+                "user_ids": [13],
+                "committee_id": 2,
+            },
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists("meeting/2", {"committee_id": 2, "user_ids": [13]})
```
Comment (r-peschke) [inline tests/system/action/meeting/test_clone.py:290]: Leave the asserts, they have to true without setting them before

**Review by r-peschke — CHANGES_REQUESTED** [review]
Lets have a look together on the tests

File: tests/system/action/meeting/test_clone.py (hunk 4 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py]: Done.

File: tests/system/action/meeting/test_clone.py (hunk 5 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:327]: Update the test and add an additional test.

File: tests/system/action/meeting/test_clone.py [hunk 6]
```diff
@@ -259,11 +261,82 @@ def test_clone_user_ids_and_admin_ids(self) -> None:
...
+        self.assert_model_exists(
+            "committee/2", {"user_ids": [13], "organization_id": 1, "meeting_ids": [2]}
+        )
+        self.assert_model_exists(
+            "user/13",
+            {
+                "username": "user_from_new_committee",
+                "committee_ids": [2],
```
Comment (r-peschke) [inline tests/system/action/meeting/test_clone.py:302]: You should assert, that the user belongs to group/2 of meeting/2 (I assume the cloned objects in both cases have the id 2) and group/2 having the user/13 as member.

File: tests/system/action/meeting/test_clone.py [hunk 7]
```diff
@@ -259,11 +261,82 @@ def test_clone_user_ids_and_admin_ids(self) -> None:
...
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists("meeting/2", {"committee_id": 2, "user_ids": [13]})
+        self.assert_model_exists(
+            "committee/2", {"user_ids": [13], "organization_id": 1, "meeting_ids": [2]}
+        )
+        self.assert_model_exists(
+            "user/13",
+            {
+                "username": "user_from_new_committee",
```
Comment (r-peschke) [inline tests/system/action/meeting/test_clone.py:334]: also check the group of user. To check this 100% check that it is the default group of the new meeting.

**Review by r-peschke — CHANGES_REQUESTED** [review]

File: tests/system/action/meeting/test_clone.py (hunk 6 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:316]: Done.

File: tests/system/action/meeting/test_clone.py (hunk 7 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:354]: Done.

File: tests/system/action/meeting/test_clone.py (hunk 6 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:316]: In the meeting 2 it is group 3 is the default group. group 2 is the admin group in meeting 1.

File: tests/system/action/meeting/test_clone.py (hunk 7 above)
Comment (reiterl) [reply tests/system/action/meeting/test_clone.py:354]: Used the link to check that group 3 is the default group of meeting 2.

**Review by r-peschke — APPROVED** [review]

**Pull request merged**